src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,123 @@
+package duke.ui;
+
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddTaskCommand;
+import duke.command.Command;
+import duke.command.DeleteTaskCommand;
+import duke.command.ExitCommand;
+import duke.command.FindTaskCommand;
+import duke.command.GetListCommand;
+import duke.command.TaskDoneCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidTaskNoException;
+import duke.exception.InvalidTimeException;
+import duke.exception.MissingCommandDetailException;
+import duke.exception.MultipleTimeSlotsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Represents a parser that deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Returns parsed command which involves time.
+     *
+     * @param words An array of words in the command.
+     * @param isEvent Whether the command is an `event` command.
+     * @return Parsed command.
+     * @throws DukeException If command is invalid.
+     */
+    private static Command parseCommandWithTime(String[] words, boolean isEvent) throws DukeException {
+        // Determine key information based on type of the task
+        String timeFormat = isEvent ? "yyyy-MM-dd HH:mm to yyyy-MM-dd HH:mm" : "yyyy-MM-dd HH:mm";
+        String regex = isEvent ? "/at" : "/by";
+        String taskType = isEvent ? "event" : "deadline";
+
+        // Parse the command
+        if (words.length >= 2) {
+            String[] information = words[1].split(regex);
+            if (information.length == 2) {
+                try {
+                    if (isEvent) {
+                        return new AddTaskCommand(new Event(information[0], information[1]));
+                    } else {
+                        return new AddTaskCommand(new Deadline(information[0], information[1]));
+                    }
+                } catch (DateTimeParseException | ArrayIndexOutOfBoundsException e) {
+                    throw new InvalidTimeException(timeFormat);
+                }
+            } else if (information.length < 2) {
+                throw new MissingCommandDetailException("time", taskType, String.format("%s %s", regex, timeFormat));
+            } else {
+                throw new MultipleTimeSlotsException(taskType);
+            }
+        } else {
+            throw new MissingCommandDetailException("description", taskType,
+                    String.format("%s %s", regex, timeFormat));
+        }
+    }
```

Comment:
Perhaps you can try using switch case statement instead of if else + try catch conditionals to avoid deep nesting? Let me know what you think!